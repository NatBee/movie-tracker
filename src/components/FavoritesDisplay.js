import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { getFavorites } from '../apiCall';
import Movie from './Movie';

class FavoritesDisplay extends Component {
  constructor() {
    super();

    this.state = {
      favorites: []
    }
  }

  componentDidMount() {
    // this.loadFavorites();
  }

  loadFavorites = async () => {
    const userId = this.props.userId;
    const results = await getFavorites(userId);
    const favorites = await results.data;

    return favorites;
    // this.renderFavorites(favorites);
  }


  renderFavorites = () => {
    const favorites = this.loadFavorites();
    return favorites.map( movie => {
      return (
        <Movie key={ movie.id }
               poster={ movie.poster_path }
               title={ movie.title }
               rating={ movie.popularity }
               overview={ movie.overview }
               releaseDate={ movie.release_date }
               id={ movie.id } />
      ) 
    })
    // return !favorites ? <p>No Favorites</p> : <p>Favorites!</p>
  }

  render() {
    return (
      <div>
        { 
          this.props.userId &&
          this.renderFavorites() 
        }
      </div>
    );
  }
}

export const mapStateToProps = (store) => ({
  userId: store.current_user.id
})

export default connect(mapStateToProps, null)(FavoritesDisplay);