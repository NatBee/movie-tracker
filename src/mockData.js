export const mockData = [{
  "vote_count": 20,
  "id": 337167,
  "video": false,
  "vote_average": 6.4,
  "title": "Fifty Shades Freed",
  "popularity": 45.670468,
  "poster_path": "/iUDFe6B6bFPOAE13uoqH4eEU1cF.jpg",
  "original_language": "en",
  "original_title": "Fifty Shades Freed",
  "genre_ids": [18, 10749],
  "backdrop_path": "/iuyE1qy9HoTsMg0DHB1KJxM94DJ.jpg",
  "adult": false,
  "overview": "The third installment of the Fifty Shades Trilogy.",
  "release_date": "2018-02-07"
}];