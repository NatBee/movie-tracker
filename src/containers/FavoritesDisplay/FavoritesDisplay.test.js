/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import { FavoritesDisplay, mapStateToProps } from './FavoritesDisplay';
import mockData from '../../mockData';

describe('FavoritesDisplay', () => {
  let wrapper;

  beforeEach(() => {
    // const mockProps = {
    //   userId: 2,
    //   favorites: mockData
    // }
    // let mockLoadFavorites = jest.fn();
    // let mockGetFavorites = jest.fn();
    // let mockRenderFavorites = jest.fn();
    // window.fetch = jest.fn();

        // loadFavorite={ mockLoadFavorites }
        //                getFavorites={ mockGetFavorites }
        //                renderFavorites={ mockRenderFavorites }/>,

      // { disableLifecycleMethods: true }
    
    wrapper = shallow(<FavoritesDisplay />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map to the store correctly', () => {
    const mockId = 0;
    const mockStore = {
      current_user: { 
        id: mockId 
      }
    }
    const mapped = mapStateToProps(mockStore)

    expect(mapped.userId).toEqual(mockStore.current_user.id);
  });
})