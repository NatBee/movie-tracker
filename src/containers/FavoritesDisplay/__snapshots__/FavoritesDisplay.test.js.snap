// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesDisplay should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoritesDisplay />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <h2
          className="favorites-alert"
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "favorites-alert",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <h2
            className="favorites-alert"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "favorites-alert",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
